internal/pipeline/pipeline.go

```diff
@@ -60,6 +73,69 @@ type Pipeline struct {
 	Validators []Function `yaml:"validators,omitempty"`
 }
 
+// String returns the string representation of Pipeline
+func (p *Pipeline) String() (string, error) {
+	b, err := yaml.Marshal(p)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to convert pipeline to string")
+	}
+	return string(b), nil
+}
+
+// New returns a pointer to a new default Pipeline.
+// The default Pipeline should be:
+// apiVersion: kpt.dev/v1alpha1
+// kind: Pipeline
+// metadata:
+//   name: pipeline
+// sources:
+//   - '.*'
+func New() *Pipeline {
+	p := &Pipeline{}
+	p.APIVersion = kptAPIVersion
+	p.Kind = pipelineKind
+	p.Name = defaultName
+	p.Sources = defaultSources
+	return p
+}
+
+// FromBytes returns a Pipeline parsed from bytes
+func FromBytes(b []byte) (*Pipeline, error) {
+	p := New()
+	err := yaml.Unmarshal(b, p)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to construct pipeline from bytes")
+	}
+	return p, nil
+}
+
+// FromString returns a Pipeline parsed from string
+func FromString(s string) (*Pipeline, error) {
+	return FromBytes([]byte(s))
+}
```

Comment:
I would suggest keeping `FromString` and `FromBytes` as internal methods because our use-cases mostly involves constructing pipeline either from a file or max from a reader. And if at anyplace we have bytes/string, it is trivial to construct io.Reader from string/bytes. This also reduces extra tests. 

We should start with minimum API surface for a package and gradually add things as we learn more.